crates/turborepo-lib/src/config/override_env.rs

```diff
@@ -0,0 +1,493 @@
+            true => Some(UIMode::Stream),
+            false => None,
+        }
+    }
+}
+
+impl<'a> ResolvedConfigurationOptions for OverrideEnvVars<'a> {
+    fn get_configuration_options(
+        &self,
+        _existing_config: Option<&ConfigurationOptions>,
+    ) -> Result<ConfigurationOptions, Error> {
+        let output = ConfigurationOptions {
+            team_id: self.output.team_id.clone(),
+            token: self.output.token.clone(),
+            team_slug: self.output.team_slug.clone(),
+            ui: self.ui(),
+            api_url: None,
+            ..Default::default()
+        };
+        Ok(output)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use lazy_static::lazy_static;
+
+    use super::*;
+
+    lazy_static! {
+        static ref VERCEL_ARTIFACTS_OWNER: String = String::from("valueof:VERCEL_ARTIFACTS_OWNER");
+        static ref VERCEL_ARTIFACTS_TOKEN: String = String::from("valueof:VERCEL_ARTIFACTS_TOKEN");
+        static ref TURBO_TEAMID: String = String::from("valueof:TURBO_TEAMID");
+        static ref TURBO_TEAM: String = String::from("valueof:TURBO_TEAM");
+        static ref TURBO_TOKEN: String = String::from("valueof:TURBO_TOKEN");
+    }
+
+    struct TestCase {
+        input: Input,
+        output: Output,
+        reason: &'static str,
+    }
+
+    impl TestCase {
+        fn new() -> Self {
+            Self {
+                input: Input::new(),
+                output: Output::new(),
+                reason: "missing",
+            }
+        }
+
+        fn reason(mut self, reason: &'static str) -> Self {
+            self.reason = reason;
+            self
+        }
+
+        #[allow(non_snake_case)]
+        fn VERCEL_ARTIFACTS_OWNER(mut self) -> Self {
+            self.input.VERCEL_ARTIFACTS_OWNER = Some(VERCEL_ARTIFACTS_OWNER.clone());
+            self
+        }
+
+        #[allow(non_snake_case)]
+        fn VERCEL_ARTIFACTS_TOKEN(mut self) -> Self {
+            self.input.VERCEL_ARTIFACTS_TOKEN = Some(VERCEL_ARTIFACTS_TOKEN.clone());
+            self
+        }
+
+        #[allow(non_snake_case)]
+        fn TURBO_TEAMID(mut self) -> Self {
+            self.input.TURBO_TEAMID = Some(TURBO_TEAMID.clone());
+            self
+        }
+
+        #[allow(non_snake_case)]
+        fn TURBO_TEAM(mut self) -> Self {
+            self.input.TURBO_TEAM = Some(TURBO_TEAM.clone());
+            self
+        }
+
+        #[allow(non_snake_case)]
+        fn TURBO_TOKEN(mut self) -> Self {
+            self.input.TURBO_TOKEN = Some(TURBO_TOKEN.clone());
+            self
+        }
+
+        fn team_id(mut self, value: String) -> Self {
+            self.output.team_id = Some(value);
+            self
+        }
+
+        fn team_slug(mut self, value: String) -> Self {
+            self.output.team_slug = Some(value);
+            self
+        }
+
+        fn token(mut self, value: String) -> Self {
+            self.output.token = Some(value);
+            self
+        }
```

Comment:
If you follow the above suggestion you can change this to also just take `&str`
```suggestion
        fn token(mut self, value: &str) -> Self {
            self.output.token = Some(value.into());
            self
        }
```

Reply:
!!!!!!!!!! AWESOME

cffcb17fa23f293f483b45695304064813db9007

that's _EXACTLY_ the API I was hoping to achieve in the first place.  thanks for walking me through that!!